external_server/models/devices.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+import dataclasses
+
+from InternalProtocol_pb2 import Device as _Device  # type: ignore
+from InternalProtocol_pb2 import DeviceStatus as _DeviceStatus  # type: ignore
+
+
+def device_status(device: _Device, status_data: bytes = b"") -> _DeviceStatus:
+    return _DeviceStatus(device=device, statusData=status_data)
+
+
+@dataclasses.dataclass(frozen=True)
+class DevicePy:
+    """This class represents a device connected to the external server.
+
+    It contains the device identifiers and methods for assessing two devices are identical.
+    """
+
+    module_id: int
+    type: int
+    role: str
+    name: str
+    priority: int
+
+    def __eq__(self, other: object):
+        if isinstance(other, DevicePy):
+            return (
+                self.module_id == other.module_id
+                and self.type == other.type
+                and self.role == other.role
+            )
+        elif isinstance(other, _Device):
+            return (
+                self.module_id == other.module
+                and self.type == other.deviceType
+                and self.role == other.deviceRole
+            )
+        else:
+            return False
+
+    def to_device(self) -> _Device:
+        """Converts the DevicePy instance to the protobuf Device message."""
+        return _Device(
+            module=self.module_id,
+            deviceType=self.type,
+            deviceRole=self.role,
+            deviceName=self.name,
+            priority=self.priority,
+        )
+
+    @staticmethod
+    def from_device(device: _Device) -> DevicePy:
+        """Creates a DevicePy instance from the protobuf Device message."""
+        return DevicePy(
+            module_id=device.module,
+            type=device.deviceType,
+            role=device.deviceRole,
+            name=device.deviceName,
+            priority=device.priority,
+        )
+
+
+@dataclasses.dataclass
+class KnownDevices:
+    """This class manages two parallel lists of DevicePy instances representing
+    devices communicating with the external server.
```

Comment:
Also explain what is connected and what is not connected device.

Reply:
Added to docstring.